symplyphysics/laws/electricity/circuits/butterworth_filter_order_from_distortion_and_frequencies.py

```diff
@@ -0,0 +1,47 @@
+from math import ceil
+from sympy import (Eq, solve, log)
+from symplyphysics import (units, Quantity, Symbol, print_expression, validate_input,
+    validate_output, dimensionless, convert_to_float)
+
+# Description
+## The approximation of the power transmission coefficient of a normalized low-pass filter is given by approximating functions of the order of n.
+## The Butterworth filter is described by the function frequency^n.
+## Bandwidth distortion determines the maximum distortion in the bandwidth. In other words, to determine the level of ripples in the bandwidth.
+## The band-stop distortion distortion determines the level of distortion in the band-stop.
+
+## Law is: N = ln(es / ep) / ln(fs, fp), where
+## N - filter order,
+## es - band-stop distortion,
+## ep - bandwidth distortion,
+## fs - frequency of the band-stop (the frequency that corresponds to the maximum band-stop distortion),
+## fp - cutoff frequency (the frequency that corresponds to the maximum bandwidth distortion).
+
+filter_order = Symbol("filter_order", dimensionless)
+
+bandwidth_distortion = Symbol("bandwidth_distortion", dimensionless)
+band_stop_distortion = Symbol("band_stop_distortion", dimensionless)
+cutoff_frequency = Symbol("cutoff_frequency", units.frequency)
+band_stop_frequency = Symbol("band_stop_frequency", units.frequency)
+
+law = Eq(filter_order, log(band_stop_distortion / bandwidth_distortion) / log(band_stop_frequency / cutoff_frequency))
+
+
+def print_law() -> str:
+    return print_expression(law)
+
+
+@validate_input(bandwidth_distortion_=bandwidth_distortion,
+    band_stop_distortion_=band_stop_distortion,
+    band_stop_frequency_=band_stop_frequency,
+    cutoff_frequency_=cutoff_frequency)
+@validate_output(filter_order)
+def calculate_order(bandwidth_distortion_: float, band_stop_distortion_: float, band_stop_frequency_: Quantity,
+    cutoff_frequency_: Quantity) -> int:
+    result_expr = solve(law, filter_order, dict=True)[0][filter_order]
+    result_expr = result_expr.subs({
+        bandwidth_distortion: bandwidth_distortion_,
+        band_stop_distortion: band_stop_distortion_,
+        band_stop_frequency: band_stop_frequency_,
+        cutoff_frequency: cutoff_frequency_,
+    })
+    return ceil(convert_to_float(result_expr))
```

Comment:
Could you explain `ceil` here? Why not `floor`?

Reply:
> Could you explain `ceil` here? Why not `floor`?

Because you can only implement an integer-order filter. And the formula itself returns the minimum order of the filter, which will have the specified parameters (frequencies and distortions). Therefore, you need to round it up.

Reply:
Add small comment in the code.